src/views/dateTimeInput.component.tsx

```diff
@@ -30,49 +30,51 @@ export function updateFilter({
   fromDateOrToDateChanged,
   onChange,
 }: UpdateFilterParams): void {
-  if (
-    date &&
-    isValid(date) &&
-    !datesEqual(date, otherDate) &&
-    (!prevDate || !datesEqual(date, prevDate))
-  ) {
-    const validFromDate =
-      fromDateOrToDateChanged === 'fromDate' &&
-      (!otherDate || isBefore(date, otherDate));
-    const validToDate =
-      fromDateOrToDateChanged === 'toDate' &&
-      (!otherDate || !isBefore(date, otherDate));
-
-    if (validFromDate || validToDate) {
-      onChange(fromDateOrToDateChanged, format(date, 'yyyy-MM-dd HH:mm:ss'));
-    }
-  } else if (!date) {
-    onChange(fromDateOrToDateChanged);
+  if (datesEqual(date, prevDate)) return; // Do nothing if the date is unchanged
+
+  if (fromDateOrToDateChanged === 'fromDate') {
+    if (date && otherDate && !isBefore(date, otherDate)) return; // Do nothing if fromDate is after toDate
+    onChange({
+      fromDate: date ? format(date, 'yyyy-MM-dd HH:mm:ss') : undefined,
+      toDate: otherDate ? format(otherDate, 'yyyy-MM-dd HH:mm:ss') : undefined,
+    });
+  } else {
+    if (date && otherDate && isBefore(date, otherDate)) return; // Do nothing if toDate is before fromDate
+    onChange({
+      fromDate: otherDate
+        ? format(otherDate, 'yyyy-MM-dd HH:mm:ss')
+        : undefined,
+      toDate: date ? format(date, 'yyyy-MM-dd HH:mm:ss') : undefined,
+    });
   }
 }
 
 export interface DateTimeFilterProps {
-  onChange: (range: 'fromDate' | 'toDate', date?: string) => void;
-  receivedFromDate?: string;
-  receivedToDate?: string;
+  onChange: (value: { fromDate?: string; toDate?: string }) => void;
+  value: { fromDate?: string; toDate?: string };
 }
 
 export const DateTimeFilter = (
   props: DateTimeFilterProps
 ): React.ReactElement => {
-  const { onChange, receivedFromDate, receivedToDate } = props;
-
   const [fromDate, setFromDate] = React.useState<Date | null>(
-    receivedFromDate ? parseISO(receivedFromDate) : null
+    props.value?.fromDate ? new Date(props.value.fromDate) : null
   );
   const [toDate, setToDate] = React.useState<Date | null>(
-    receivedToDate ? parseISO(receivedToDate) : null
+    props.value?.toDate ? new Date(props.value.toDate) : null
   );
 
   const [popupOpen, setPopupOpen] = React.useState<boolean>(false);
 
   const invalidDateRange = fromDate && toDate && isBefore(toDate, fromDate);
 
+  const initialFromDate = props.value?.fromDate
+    ? new Date(props.value.fromDate)
+    : null;
+  const initialToDate = props.value?.toDate
+    ? new Date(props.value.toDate)
+    : null;
+
```

Comment:
Can we move these earlier and use these to initialise `fromDate` and `toDate`? Just reduces some repetition